include/fiction/algorithms/simulation/sidb/detect_bdl_wires.hpp

```diff
@@ -0,0 +1,681 @@
+//
+// Created by Jan Drewniok on 15.07.24.
+//
+
+#ifndef FICTION_DETECT_BDL_WIRES_HPP
+#define FICTION_DETECT_BDL_WIRES_HPP
+
+#include "fiction/algorithms/path_finding/distance.hpp"
+#include "fiction/algorithms/simulation/sidb/detect_bdl_pairs.hpp"
+#include "fiction/technology/cell_ports.hpp"
+#include "fiction/technology/cell_technologies.hpp"
+#include "fiction/traits.hpp"
+
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <optional>
+#include <set>
+#include <vector>
+
+namespace fiction
+{
+
+/**
+ * An enumeration of the selection of different types of wires.
+ */
+enum class bdl_wire_selection : std::uint8_t
+{
+    /**
+     * Select all BDL wires.
+     */
+    ALL,
+    /**
+     * Select only BDL wires that start with input cells.
+     */
+    INPUT,
+    /**
+     * Select only BDL wires that end with output cells.
+     */
+    OUTPUT
+};
+
+/**
+ * This struct encapsulates parameters used for detecting BDL wires.
+ */
+struct detect_bdl_wires_params
+{
+    /**
+     * A distance threshold, which is used to determine if two pairs of BDLs are part of the same wire. (unit: nm).
+     */
+    double threshold_bdl_interdistance = 2.0;
+    /**
+     * Parameters for the `detect_bdl_pairs` algorithm.
+     */
+    detect_bdl_pairs_params bdl_pairs_params{};
+};
+
+/**
+ * This struct encapsulates a vector of `bdl_pair` objects, representing the pairs of SiDBs in the BDL wire.
+ *
+ * @tparam CellType Cell type.
+ */
+template <typename Lyt>
+struct bdl_wire
+{
+    /**
+     * Vector of BDL pairs representing the wire.
+     */
+    std::vector<bdl_pair<cell<Lyt>>> pairs{};
+
+    /**
+     * Port of the BDL wire.
+     */
+    port_direction port{port_direction::NONE};
+
+    /**
+     * Starting BDL pair of the wire.
+     */
+    std::optional<bdl_pair<cell<Lyt>>> first_bdl_pair_wire{};
```

Comment:
I would argue the naming here is not optimal as right now it sounds like this is the first wire consisting of BDLs. Rather than the first BDL pair of a wire.